union-ai-admin/aws/script/generate.py

```diff
@@ -183,39 +185,90 @@ def create_read_policy(role_type):
+                        Sub("arn:aws:iam::${AWS::AccountId}:role/*adminflyterole*"),
+                        Sub("arn:aws:iam::${AWS::AccountId}:role/*fluentbitrole*"),
                     ],
                 ),
                 Statement(
                     Effect=Allow,
                     Action=[
-                        Action("iam", "ListOpenIDConnectProviderTags"),
                         Action("iam", "ListPolicyVersions"),
                         Action("iam", "ListPolicyTags"),
                         Action("iam", "ListRoleTags"),
-                        Action("iam", "ListInstanceProfilesForRole"),
-                        Action("iam", "ListInstanceProfileTags"),
                         Action("iam", "ListRoles"),
+                        Action("iam", "ListInstanceProfilesForRole"),
                         Action("iam", "ListRolePolicies"),
                         Action("iam", "ListAttachedRolePolicies"),
                     ],
+                    Resource=[
+                        Sub("arn:aws:iam::${AWS::AccountId}:policy/opta-*"),
+                        Sub("arn:aws:iam::${AWS::AccountId}:policy/unionai-*"),
+                        Sub("arn:aws:iam::${AWS::AccountId}:policy/*userflyterole*"),
+                        Sub("arn:aws:iam::${AWS::AccountId}:policy/*adminflyterole*"),
+                        Sub("arn:aws:iam::${AWS::AccountId}:policy/*fluentbitrole*"),
+                        Sub("arn:aws:iam::${AWS::AccountId}:policy/*fluentbitpolicy*"),
+                        Sub("arn:aws:iam::${AWS::AccountId}:role/*AWSService*"),
+                        Sub("arn:aws:iam::${AWS::AccountId}:role/opta-*"),
+                        Sub("arn:aws:iam::${AWS::AccountId}:role/unionai-*"),
+                        Sub("arn:aws:iam::${AWS::AccountId}:role/*userflyterole*"),
+                        Sub("arn:aws:iam::${AWS::AccountId}:role/*adminflyterole*"),
+                        Sub("arn:aws:iam::${AWS::AccountId}:role/*fluentbitrole*"),
+                    ],
+                ),
+                Statement(
+                    Effect=Allow,
+                    Action=[
+                        Action("iam", "ListOpenIDConnectProviderTags"),
+                        Action("iam", "ListInstanceProfileTags"),
+                    ],
                     Resource=[
                         Sub("arn:aws:iam::${AWS::AccountId}:oidc-provider/*"),
-                        Sub("arn:aws:iam::${AWS::AccountId}:policy/*"),
-                        Sub("arn:aws:iam::${AWS::AccountId}:role/*"),
                         Sub("arn:aws:iam::${AWS::AccountId}:instance-profile/*"),
                     ],
                 ),
+                Statement(
+                    Effect=Allow,
+                    Action=[
+                        Action("ec2", "DescribeVpcAttribute"),
+                        Action("ec2", "AssociateAddress"),
+                        Action("ec2", "DisassociateAddress"),
```

Comment:
Is this correct? Should the reader policy have permissions to AssociateAddress and DisassociateAddress?

Reply:
This section of code comes the from the function defined with `def create_read_policy(role_type):`  This policy is attached the unionai-reader role and the other roles.